### ISSUE TITLE
VPN Container Restart, Child Containers don't reconnect.

### ISSUE THREAD

**Issue by kkniffin** [haugene/docker-transmission-openvpn#103]
I am experiencing an issue where I have a parent container which houses the docker-transmission-openvpn image. I then have child containers that have their network_mode set to use the vpn service. Everything works great until the vpn service fails and restarts. It seems after the restart that the child containers no longer link back up to the vpn containers networking? Is anyone else experiencing this? Should this be the case or is there something I am missing? I have set the various depends_on selections in docker_compose. Here is my docker-compose file. https://github.com/kkniffin/usb-mediaserver/blob/master/docker-compose.yml

I am running the latest version of Debian Jessie and have tried various versions of docker up to the latest experimental rc4 that I am on now:

Distributor ID: Debian
Description: Debian GNU/Linux 8.5 (jessie)
Release: 8.5
Codename: jessie

Client:
Version: 1.12.0-rc4
API version: 1.24
Go version: go1.6.2
Git commit: e4a0dbc
Built: Wed Jul 13 03:37:36 2016
OS/Arch: linux/amd64
Experimental: true

Server:
Version: 1.12.0-rc4
API version: 1.24
Go version: go1.6.2
Git commit: e4a0dbc
Built: Wed Jul 13 03:37:36 2016
OS/Arch: linux/amd64
Experimental: true

If I try to re-connect to one of the child containers I receive the following error on all of them:

rpc error: code = 2 desc = oci runtime error: exec failed: container_linux.go:1215: running lstat on namespace path "/proc/1225/ns/net" caused "lstat /proc/1225/ns/net: no such file or directory"

Appreciate any guidance or help in troubleshooting.

**Comment by haugene** [comment]
I think I've experienced something similar using container linking in some dependant services. But that is not with the networking that you describe here and not with this container.

But I also have a feeling that this kind of dependencies are not handled, and I've had to restart/recreate the containers that depend on it. But I don't know how to fix it other than having a script check for some log statements or whatever and then restart stuff after the container has restarted.

Let me know if you figure out a more Docker-native way of resolving the issue :)
